gateway/server.go

```diff
@@ -282,6 +285,12 @@ func (gw *Gateway) getApiSpec(apiID string) *APISpec {
 	return spec
 }
 
+func (gw *Gateway) removeApiById(apiID string) {
+	gw.apisMu.RLock()
+	delete(gw.apisByID, apiID)
```

Comment:
As you're modifying data, should use Lock/Unlock, and not RLock/RUnlock.

Reply:
This new func isn't used anywhere and is reported in golangcilint